routers/auth.js

```diff
@@ -0,0 +1,297 @@
+// - checking if a token is (still) valid
+router.get("/me", authMiddleware, async (req, res) => {
+  const posts = await Post.findAll({
+    where: { userId: req.user.id },
+    include: { model: Image },
+    order: [["createdAt", "DESC"]],
+  });
+  const likes = await Like.findAll({
+    where: { userId: req.user.id },
+    order: [["createdAt", "DESC"]],
+  });
+  // don't send back the password hash
+  delete req.user.dataValues["password"];
+  res.status(200).send({ ...req.user.dataValues, posts, likes });
+});
+router.get("/:userId", async (req, res) => {
+  const { userId } = req.params;
+  const posts = await Post.findAll({
+    where: { userId: userId },
+    include: { model: Image },
+    order: [["createdAt", "DESC"]],
+  });
+
+  res.status(200).send(posts);
+});
+
+// update personal infor: name, description
+router.patch("/:userId", authMiddleware, async (req, res) => {
+  const updatedUser = await User.findByPk(req.params.userId);
+  const { name, description } = req.body;
+
+  await updatedUser.update({ name, description });
+  res.status(200).send(updatedUser);
+});
+
+// update new profile image
+router.patch("/:userId/profilePic", authMiddleware, async (req, res) => {
+  try {
+    const updatedUser = await User.findByPk(req.params.userId);
+    const { profile_pic } = req.body;
+    const uploadedResponse = await cloudinary.uploader.upload(profile_pic, {
+      upload_preset: "pets-dev",
+    });
+    await updatedUser.update({ profile_pic: uploadedResponse.public_id });
+    res.status(200).send(updatedUser);
+  } catch (error) {
+    console.log(error);
+    return res.status(400).send({ message: "Something went wrong, sorry" });
+  }
+});
+
+// like a post
+router.post("/post/:postId/like", authMiddleware, async (req, res) => {
+  try {
+    const { userId } = req.body;
+    const postId = req.params.postId;
+    const like = await Like.findOne({
+      where: { userId: userId, postId: postId },
+    });
+
+    const post = await Post.findOne({
+      where: { id: postId },
+      include: [
+        { model: Image },
+        { model: User, attributes: ["name", "profile_pic"] },
+        {
+          model: Comment,
+          include: [{ model: User, attributes: ["name", "profile_pic"] }],
+          order: [["createdAt", "DESC"]],
+        },
+      ],
+    });
+    if (post === null) {
+      res.status(400).send({ error: "Post does not exist" });
+    } else {
+      if (like === null) {
+        const newLike = await Like.create({
+          userId: userId,
+          postId: postId,
+        });
+        post.likes_num++;
+        await post.update({ likes_num: post.likes_num });
+        return res.status(200).send({ message: "Like successfully", post });
+      } else {
+        return res.status(400).send({ error: "You already liked this post" });
+      }
+    }
+  } catch (error) {
+    console.log(error);
+  }
+});
+
+router.get("/post/:postId/allLikes", authMiddleware, async (req, res) => {
```

Comment:
i think all this stuff deserves its own `router`, doesn't really have a lot to do with `auth`. Move them over to a `post` router or similar